Circular import fix

MoodyMan04 opened the pull request.

The-8th-wonder committed:
Update player.py

The-8th-wonder committed:
Merge branch 'master' into player

The-8th-wonder committed:
Update player.py

The-8th-wonder committed:
Update player.py

The-8th-wonder committed:
Update player.py

MoodyMan04 committed:
Create test_initialization.py

MoodyMan04 committed:
item durability + some unit tests

item durability is able to be set to None if infinite durability

issues to look into: circular import,
needed unit tests, etc.

MoodyMan04 committed:
Revert "item durability + some unit tests"

This reverts commit cb43a4965c011ad77c6ec6d5747bc9740786e1f5.

MoodyMan04 committed:
Merge branch 'unit_tests' into circular_import_fix

MoodyMan04 committed:
unit tests + changes

fixed circular import with:
occupiable interface
vector class

unit tests added:
initialization
avatar
tile
occupiable_station

changes:
avatar
enums
tile
occupiable_station
station
vector

MoodyMan04 committed:
update python-app to 3.11

MoodyMan04 committed:
added vector to test_initialization

ESatrom commented on game/common/enums.py:
```diff
@@ -18,6 +18,7 @@ class ObjectType(Enum):
     TILE = auto()
     WALL = auto()
     ITEM = auto()
+    ITERACTABLE = auto()
```
Was this meant to be `INTERACTABLE`?

ESatrom reviewed (approved):
Looks good

I noticed the potential typo enum of iteractable doesn't seem to be used anywhere, even the test cases. I'm really not sure what's going on with that one, but everything else looks good.

JeanVArdent commented on game/common/map/occupiable.py:
```diff
@@ -0,0 +1,34 @@
+class Occupiable(GameObject):
+    def __init__(self, occupied_by: GameObject = None, **kwargs):
+        super().__init__()
+        self.object_type: ObjectType = ObjectType.ITERACTABLE
```
Same thing that Elijah said

JeanVArdent commented on game/common/map/tile.py:
```diff
@@ -1,40 +1,25 @@
     def from_json(self, data: dict) -> Self:
         super().from_json(data)
-        occupied_by: GameObject|None = data['occupied_by']
+        occupied_by = data['occupied_by']
```
Type hint dict

JeanVArdent commented on game/common/map/tile.py:
```diff
@@ -43,6 +28,8 @@ def from_json(self, data: dict) -> Self:
                 self.occupied_by = Occupiable_Station().from_json(data['occupied_by'])
             case ObjectType.STATION:
                 self.occupied_by = Station().from_json(data['occupied_by'])
+            case ObjectType.WALL:
```
Type hint Wall

JeanVArdent commented on game/common/map/tile.py:
```diff
@@ -43,6 +28,8 @@ def from_json(self, data: dict) -> Self:
                 self.occupied_by = Occupiable_Station().from_json(data['occupied_by'])
             case ObjectType.STATION:
                 self.occupied_by = Station().from_json(data['occupied_by'])
```
Type hint Station

JeanVArdent commented on game/common/stations/occupiable_station.py:
```diff
@@ -1,24 +1,33 @@
+        # Add all possible game objects that can occupy a tile here (requires python 3.10) 
+        match occupied_by['object_type']:
+            case ObjectType.AVATAR:
+                self.occupied_by = Avatar().from_json(data['occupied_by'])
```
Type hint

JeanVArdent commented on game/common/stations/occupiable_station.py:
```diff
@@ -1,24 +1,33 @@
+            case ObjectType.AVATAR:
+                self.occupied_by = Avatar().from_json(data['occupied_by'])
+            case ObjectType.OCCUPIABLE_STATION:
+                self.occupied_by = Occupiable_Station().from_json(data['occupied_by'])
```
Type hint

JeanVArdent commented on game/common/stations/occupiable_station.py:
```diff
@@ -1,24 +1,33 @@
+            case ObjectType.OCCUPIABLE_STATION:
+                self.occupied_by = Occupiable_Station().from_json(data['occupied_by'])
+            case ObjectType.STATION:
+                self.occupied_by = Station().from_json(data['occupied_by'])
```
Type hint

JeanVArdent reviewed (changes requested):
Just a couple type hints if possible and changing the enum. Great job cranking out these unit tests

MoodyMan04 committed:
type hint, enum, and readme fix

fixed enum:

Iteractable -> Occupiable
added type hinting:

tile and occupiable_station from_json methods have more type hinting in match-case section

added to README.md:

added note about requiring Python 3.11 due to type Self

JeanVArdent reviewed (approved):
Lgtm

MoodyMan04 committed:
Merge branch 'master' into circular_import_fix
